ortools/linear_solver/xpress_interface.cc

```diff
@@ -1,1447 +1,1512 @@
+void XpressInterface::SetLpAlgorithm(int value) {
+  MPSolverParameters::LpAlgorithmValues const algorithm =
+      static_cast<MPSolverParameters::LpAlgorithmValues>(value);
+
+  int alg = 1;
+
+  switch (algorithm) {
+    case MPSolverParameters::DUAL:
+      alg = 2;
+      break;
+    case MPSolverParameters::PRIMAL:
+      alg = 3;
+      break;
+    case MPSolverParameters::BARRIER:
+      alg = 4;
+      break;
+  }
+
+  if (alg == XPRS_DEFAULTALG) {
+    SetIntegerParamToUnsupportedValue(MPSolverParameters::LP_ALGORITHM, value);
+  } else {
+    CHECK_STATUS(XPRSsetintcontrol(mLp, XPRS_DEFAULTALG, alg));
+  }
+}
+
+bool XpressInterface::ReadParameterFile(std::string const& filename) {
+  // Return true on success and false on error.
+  LOG(DFATAL) << "ReadParameterFile not implemented for XPRESS interface";
+  return false;
+}
+
+std::string XpressInterface::ValidFileExtensionForParameterFile() const {
+  return ".prm";
+}
+
+MPSolver::ResultStatus XpressInterface::Solve(MPSolverParameters const& param) {
+  int status;
+
+  // Delete chached information
+  mCstat = 0;
+  mRstat = 0;
+
+  WallTimer timer;
+  timer.Start();
+
+  // Set incrementality
+  MPSolverParameters::IncrementalityValues const inc =
+      static_cast<MPSolverParameters::IncrementalityValues>(
+          param.GetIntegerParam(MPSolverParameters::INCREMENTALITY));
+  switch (inc) {
+    case MPSolverParameters::INCREMENTALITY_OFF: {
+      Reset();  // This should not be required but re-extracting everything
+                // may be faster, so we do it.
+      break;
+    }
+    case MPSolverParameters::INCREMENTALITY_ON: {
+      XPRSsetintcontrol(mLp, XPRS_CRASH, 0);
+      break;
+    }
+  }
+
+  // Extract the model to be solved.
+  // If we don't support incremental extraction and the low-level modeling
+  // is out of sync then we have to re-extract everything. Note that this
+  // will lose MIP starts or advanced basis information from a previous
+  // solve.
+  if (!supportIncrementalExtraction && sync_status_ == MUST_RELOAD) Reset();
+  ExtractModel();
+  VLOG(1) << absl::StrFormat("Model build in %.3f seconds.", timer.Get());
+
+  // Set log level.
+  XPRSsetintcontrol(mLp, XPRS_OUTPUTLOG, quiet() ? 0 : 1);
+  // Set parameters.
+  // NOTE: We must invoke SetSolverSpecificParametersAsString() _first_.
+  //       Its current implementation invokes ReadParameterFile() which in
+  //       turn invokes XPRSreadcopyparam(). The latter will _overwrite_
+  //       all current parameter settings in the environment.
+  solver_->SetSolverSpecificParametersAsString(
+      solver_->solver_specific_parameter_string_);
+  SetParameters(param);
+  if (solver_->time_limit()) {
+    VLOG(1) << "Setting time limit = " << solver_->time_limit() << " ms.";
+    // In Xpress, a time limit should usually have a negative sign. With a
+    // positive sign, the solver will only stop when a solution has been found.
+    CHECK_STATUS(XPRSsetdblcontrol(mLp, XPRS_MAXTIME,

```

Comment:
Can this be fixed as well in this PR? https://github.com/google/or-tools/discussions/2954